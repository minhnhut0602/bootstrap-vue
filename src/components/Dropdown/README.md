# Dropdown

```vue live
<Dropdown>Default Example Usage</Dropdown>
```

## Slots

| Name           | Description | Bindings |
| -------------- | ----------- | -------- |
| dropdownToggle |             |          |
| default        |             |          |
| dropdownMenu   |             | <br/>    |
| prepend        |             |          |
| items          |             |          |
| item           |             |          |
| append         |             |          |
