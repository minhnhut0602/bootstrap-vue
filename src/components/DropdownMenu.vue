<template>
    <ul class="dropdown-menu">
        <li
            v-for="(item, index) of items"
            :key="`item-${index}`"
        >
            <hr
                v-if="item.divider"
                class="dropdown-divider"
            >

            <h6
                v-else-if="item.header"
                class="dropdown-header"
            >
                {{ item.label }}
            </h6>

            <a
                v-else
                class="dropdown-item"
                :class="{active: item.active}"
                href="#"
                @click.prevent="clickItem(item)"
            >
                {{ item.label }}
            </a>
        </li>
    </ul>
</template>

<script lang="ts" setup>
import {Item, dropdownItemsProps} from '@/composables/useDropdownItems';
import {defineEmits, defineProps} from 'vue';

defineProps(dropdownItemsProps);

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
