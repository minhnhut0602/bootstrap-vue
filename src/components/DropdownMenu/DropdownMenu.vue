<template>
    <ul class="dropdown-menu">
        <DropdownMenuItem
            v-for="(item, index) of items"
            :key="`item-${index}`"
            v-bind="item"
        />
    </ul>
</template>

<script lang="ts" setup>
import {Item, dropdownItemsProps} from '../../composables/useDropdownItems';
import DropdownMenuItem from '../DropdownMenuItem';

defineProps(dropdownItemsProps);

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
