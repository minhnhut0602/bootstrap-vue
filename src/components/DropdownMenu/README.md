# DropdownMenu

```vue live
<DropdownMenu>Default Example Usage</DropdownMenu>
```

## Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| prepend |             |          |
| items   |             | <br/>    |
| item    |             | <br/>    |
| append  |             |          |

## Events

| Event name | Properties | Description |
| ---------- | ---------- | ----------- |
| click:item |            |
