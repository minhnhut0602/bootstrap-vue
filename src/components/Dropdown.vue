<template>
    <div class="dropdown">
        <Btn
            class="dropdown-toggle"
            data-bs-toggle="dropdown"
            :size="size"
            :variant="variant"
        >
            {{ text }}
        </Btn>

        <DropdownMenu
            :items="items"
            @click="clickItem($event)"
        />
    </div>
</template>

<script lang="ts" setup>
import {Item, dropdownItemsProps} from '@/composables/useDropdownItems';
import {defineEmits, defineProps} from 'vue';
import Btn from '@/components/Button.vue';
import DropdownMenu from '@/components/DropdownMenu.vue';
import {sizeProps} from '@/composables/useSize';
import {variantProps} from '@/composables/useVariant';

defineProps({
    ...dropdownItemsProps,
    ...sizeProps,
    text: {
        type: String,
        required: true,
    },
    ...variantProps,
});

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
