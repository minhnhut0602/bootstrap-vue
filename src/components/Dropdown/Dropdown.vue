<template>
    <div class="dropdown">
        <Btn
            class="dropdown-toggle"
            data-bs-toggle="dropdown"
            :size="size"
            :variant="variant"
        >
            <slot />
        </Btn>

        <DropdownMenu
            :items="items"
            @click:item="clickItem($event)"
        />
    </div>
</template>

<script lang="ts" setup>
import {Item, dropdownItemsProps} from '../../composables/useDropdownItems';
import Btn from '../Button';
import DropdownMenu from '../DropdownMenu';
import {sizeProps} from '../../composables/useSize';
import {variantProps} from '../../composables/useVariant';

defineProps({
    ...dropdownItemsProps,
    ...sizeProps,
    ...variantProps,
});

const emit = defineEmits(['click:item']);

const clickItem = (item: Item) => emit('click:item', item);
</script>
