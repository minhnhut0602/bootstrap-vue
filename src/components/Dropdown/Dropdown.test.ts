import Dropdown from '.';
import DropdownMenu from '../DropdownMenu';
import {shallowMount} from '@vue/test-utils';

const items = [
    {label: 'Item 1'},
    {label: 'Item 2'},
    {label: 'Item 3'},
];

describe('template', () => {
    componentRenderTest(Dropdown, {props: {items}});

    componentSlotRenderTest(Dropdown);

    it('emits event on click dropdown menu item', async () => {
        const wrapper = shallowMount(DropdownMenu);

        wrapper.findComponent(DropdownMenu).vm.$emit('click:item', items[0]);

        expect(wrapper.emitted('click:item')[0]).toEqual([items[0]]);
    });
});
