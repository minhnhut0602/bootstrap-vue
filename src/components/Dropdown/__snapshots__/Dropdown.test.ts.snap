// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<div class=\\"dropdown\\">
  <anonymous-stub disabled=\\"false\\" outline=\\"false\\" size=\\"md\\" variant=\\"primary\\" class=\\"dropdown-toggle\\" data-bs-toggle=\\"dropdown\\"></anonymous-stub>
  <anonymous-stub items=\\"[object Object],[object Object],[object Object]\\"></anonymous-stub>
</div>"
`;
